hypothesis-python/src/hypothesis/strategies/_internal/strings.py

```diff
@@ -142,7 +143,32 @@ def filter(self, condition):
                 HypothesisWarning,
                 stacklevel=2,
             )
+
         elems = unwrap_strategies(self.element_strategy)
+
+        kwargs, pred = get_integer_predicate_bounds(condition)
+
```

Comment:
Move all of this to `ListStrategy`, so it'll be shared between all our collections types.  

And instead of modifying the instance in place, we should return a _new_ instance of `type(self)` - there might be multiple references, so in-place mutation is unsafe.